test/loader/serializeInstructionTest.cpp

```diff
@@ -666,4 +666,119 @@ TEST(SerializeInstructionTest, SerializeConstInstruction) {
   };
   EXPECT_EQ(Output, Expected);
 }
+
+TEST(SerializeInstructionTest, SerializeFuncRefControlInstruction) {
+  std::vector<uint8_t> Expected;
+  std::vector<uint8_t> Output;
+  std::vector<WasmEdge::AST::Instruction> Instructions;
+
+  // 12. Test typed function references control flow instructions.
+  //
+  //   1.  Serialize br_on_null instruction.
+  //   2.  Serialize br_on_non_null instruction.
+
+  Conf.addProposal(WasmEdge::Proposal::FunctionReferences);
+  WasmEdge::AST::Instruction Br_on_null(WasmEdge::OpCode::Br_on_null);
+  WasmEdge::AST::Instruction Br_on_non_null(WasmEdge::OpCode::Br_on_non_null);
+  WasmEdge::AST::Instruction End(WasmEdge::OpCode::End);
+  Br_on_null.getJump().TargetIndex = 0xFFFFFFFFU;
+  Instructions = {Br_on_null, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected = {
+      0x0AU,                             // Code section
+      0x0AU,                             // Content size = 10
+      0x01U,                             // Vector length = 1
+      0x08U,                             // Code segment size = 8
+      0x00U,                             // Local vec(0)
+      0xD5U,                             // OpCode Br_on_null.
+      0xFFU, 0xFFU, 0xFFU, 0xFFU, 0x0FU, // Label index.
+      0x0BU                              // Expression End.
+  };
+  EXPECT_EQ(Output, Expected);
+
+  Br_on_non_null.getJump().TargetIndex = 0xFFFFFFFFU;
+  Instructions = {Br_on_non_null, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[5] = 0xD6U; // OpCode Br_on_non_null.
+  EXPECT_EQ(Output, Expected);
+
+  // Remove FunctionReferences Proposal
+  Conf.removeProposal(WasmEdge::Proposal::FunctionReferences);
+}
+
+TEST(SerializeInstructionTest, SerializeFuncRefCallInstruction) {
```

Comment:
These are call instructions. Move them into the `SerializeCallInstruction` test.